### PR TITLE
feat: Utrecht select component with tests

### PR DESCRIPTION
**Pull request by Robbert**

### CODE REVIEW

File: src/components/utrecht/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          className
+        )}
+      >
+        {children}
+      </option>
+    );
+  }
+);
+
+SelectOption.displayName = "utrecht-select__option";
```
Comment (Yolijn) [inline src/components/utrecht/Select.tsx:68]: What do you think about using kebab-case for this displayName?

File: src/components/utrecht/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      >
+        {children}
+      </select>
+    );
+  }
+);
+
+Select.displayName = "utrecht-radio-button";
+
+interface SelectOptionProps extends SelectHTMLAttributes<HTMLOptionElement> {
```
Comment (Yolijn) [inline src/components/utrecht/Select.tsx:38]: Should this be SelectHTMLAttributes when it's an option element?

File: src/components/utrecht/Select.test.tsx [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+    });
+
+    expect(select).toContain(option);
+  });
+
+  describe("invalid variant", () => {
+    it("can have an invalid state", () => {
+      const { container } = render(<Select invalid />);
+
+      const select = container.querySelector(":only-child");
```
[the remark this answers is not in the document]
Comment (Yolijn) [reply src/components/utrecht/Select.test.tsx:61]: maybe by using `expect(select).toBeInvalid()` ?

File: src/components/utrecht/Select.test.tsx [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+
+      expect(select).not.toBeInvalid();
+    });
+
+    it("renders a design system BEM modifier class name", () => {
+      const { container } = render(<Select invalid />);
+
+      const select = container.querySelector(":only-child");
+
+      expect(select).toHaveClass("utrecht-select--invalid");
```
Comment (Yolijn) [inline src/components/utrecht/Select.test.tsx:79]: Should we also double check the modifier className is not set when the component is not invalid?

Comment (Yolijn) [reply src/components/utrecht/Select.test.tsx:79]: Maybe one test to validate no modifier classes are set by default?

File: src/components/utrecht/Select.test.tsx [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+      expect(select).toBeInTheDocument();
+    });
+
+    it("is not invalid by default", () => {
+      const { container } = render(<SelectOption />);
+
+      const select = container.querySelector(":only-child");
+
+      expect(select).not.toBeInvalid();
+      expect(select).not.toHaveClass("utrecht-select__option--invalid");
```
Comment (Yolijn) [inline src/components/utrecht/Select.test.tsx:363]: I like this way of checking a modifier class is not present if the invalid attribute is not set. This is what I would also like for the disabled etc. properties of the Select component above

File: src/components/utrecht/Select.tsx (hunk 2 above)
Comment (Robbert) [reply src/components/utrecht/Select.tsx:38]: Will fix this to use `OptionHTMLAttributes`

File: src/components/utrecht/Select.test.tsx [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+
+  it("can have a custom class name", () => {
+    const { container } = render(<Select className="ballot-box" />);
+
+    const select = container.querySelector(":only-child");
+
+    expect(select).toHaveClass("ballot-box");
+  });
+
+  describe.skip("change event", () => {
```
[the remark this answers is not in the document]
Comment (Robbert) [reply src/components/utrecht/Select.test.tsx]: Didn't have time yet to investigate how to select another option. Now I've installed a new testing-library component so I can use [`selectOptions`](https://testing-library.com/docs/user-event/utility#-selectoptions-deselectoptions) for this purpose, and I have enabled these tests.

**Review by Yolijn — APPROVED** [review]

**Pull request merged**